syntax = "proto3";
package db_package;

service ProxyDatabaseCommunication{
  //Two obligated functions
  rpc AppendEntries(LogEntry) returns (LogEntryAnswer) {}
  rpc requestVote(voteRequest) returns (voteRequestAnswer) {}

}

message voteRequest{
  int64 term = 1;
  int64 candidateId =2;
  int64 lastLogIndex=3;
  int64 lastLogTerm=4;
}

message voteRequestAnswer{
  int64 term = 1;
  bool voteGranted = 2;
}

message LogEntry{
  int64 term = 1;
  int64 leaderId = 2;
  int64 prevLogIndex = 3;
  int64 prevLogTerm = 4;
  repeated string entries = 5;
  int64 leaderCommit = 6;
}

message LogEntryAnswer{
  int64 term = 1;
  bool success = 2;
}